src/canvas/heatmapFeature.js

```diff
@@ -116,19 +124,55 @@ var canvas_heatmapFeature = function (arg) {
    * @protected
    */
   ////////////////////////////////////////////////////////////////////////////
-  this._colorize = function (pixels, gradient) {
-    var i, j;
-    for (i = 0; i < pixels.length; i += 4) {
-      // Get opacity from the temporary canvas image,
-      // then multiply by 4 to get the color index on linear gradient
-      j = pixels[i + 3] * 4;
-      if (j) {
-        pixels[i] = gradient[j];
-        pixels[i + 1] = gradient[j + 1];
-        pixels[i + 2] = gradient[j + 2];
-        pixels[i + 3] = m_this.style('opacity') * gradient[j + 3];
+  this._colorize = function (context2d, width, height, imageData, gradient) {
+    var isLittleEndian = true, i, j, index;
+
+    // Determine whether Uint32 is little- or big-endian.
+    if (!m_typedBuffer || (m_typedBuffer.length !== imageData.data.length)) {
+      m_typedBuffer = new ArrayBuffer(imageData.data.length);
+      m_typedClampedBuffer = new Uint8ClampedArray(m_typedBuffer);
+      m_typedBufferData = new Uint32Array(m_typedBuffer);
+    }
+
+    m_typedBufferData[1] = 0x0a0b0c0d;
+
+    isLittleEndian = true;
+    if (m_typedBuffer[4] === 0x0a &&
+        m_typedBuffer[5] === 0x0b &&
+        m_typedBuffer[6] === 0x0c &&
+        m_typedBuffer[7] === 0x0d) {
```

Comment:
When I try this out in the Chrome console, it doesn't work.  `m_typedBuffer` references an `ArrayBuffer`, which doesn't give me index reference access to the array.  I think this comparison needs to be on the `m_typedClampedBuffer`, not on `m_typedBuffer`.  What is here works, but only because we've only tried in on little-endian machines.